thorlcr/graph/thgraph.cpp

```diff
@@ -3221,6 +3224,9 @@ CActivityBase::CActivityBase(CGraphElementBase *_container, const StatisticsMapp
     defaultRoxieMemHeapFlags = (roxiemem::RoxieHeapFlags)container.getOptInt("heapflags", defaultHeapFlags);
     if (container.queryJob().queryUsePackedAllocators())
         defaultRoxieMemHeapFlags = (roxiemem::RoxieHeapFlags)(defaultRoxieMemHeapFlags | roxiemem::RHFpacked);
+    if (traceOptName)
+        traceActivity = getOptBool(traceOptName, false);
+    container.setDetailedTracing(traceActivity);
```

Comment:
I think this should be other way around, i.e. the container probes the option, and this (activity object) either uses the container.traceActivity flag, or copies it locally.

(see other comment re. traceOptName to container)